src/tab-container-element.ts

```diff
@@ -144,6 +89,47 @@ export class TabContainerElement extends HTMLElement {
     }
   }
 
+  handleEvent(event: Event) {
+    if (event.type === 'click') return this.#handleClick(event as MouseEvent)
+    if (event.type === 'keydown') return this.#handleKeydown(event as KeyboardEvent)
+  }
+
+  #handleKeydown(event: KeyboardEvent) {
+    const tab = (event.target as HTMLElement)?.closest?.('[role="tab"]')
+    if (!tab) return
+    const tabs = getTabs(this)
+    if (!tabs.includes(tab as HTMLElement)) return
+
+    const currentIndex = tabs.indexOf(tabs.find(e => e.matches('[aria-selected="true"]'))!)
+    const vertical = tab.closest('[role="tablist"]')?.getAttribute('aria-orientation') === 'vertical'
+    const prevTab = event.code === 'ArrowLeft' || (vertical && event.code === 'ArrowUp')
```

Comment:
Reminder for myself to check that we're setting the aria-orientation when we set horizontal or vertical.